### PR TITLE
MCOL-4530: toCppCode() method for ParseTree and TreeNode

### PR DESCRIPTION
**Pull request by mariadb-AndreyPiskunov**
For creating proper testing environment for MCOL-4530.

### CODE REVIEW

File: dbcon/execplan/constantcolumn.h [hunk 1]
```diff
@@ -72,6 +74,8 @@ class ConstantColumn : public ReturnedColumn
    * ctor
    */
   ConstantColumn(const uint64_t val, TYPE type = NUM, int8_t scale = 0, uint8_t precision = 0);  // deprecate
+
+  ConstantColumn(const string& constval, int tyoe, const string& data);
```
Comment (mariadb-LeonidFedorov) [inline dbcon/execplan/constantcolumn.h]: no need in this ctor

**Review by mariadb-LeonidFedorov — DISMISSED** [review]

File: dbcon/execplan/filter.h [hunk 2]
```diff
@@ -25,6 +25,7 @@
 #pragma once
 #include <string>
 #include <iosfwd>
+#include <unordered_set>
```
Comment (mariadb-LeonidFedorov) [inline dbcon/execplan/filter.h]: do we need it?

File: dbcon/execplan/rowcolumn.h [hunk 3]
```diff
@@ -58,7 +60,7 @@ class RowColumn : public ReturnedColumn
    */
   RowColumn(const uint32_t sessionID = 0);
   RowColumn(const RowColumn& rhs, const uint32_t sessionID = 0);
-
+  RowColumn(std::vector<SRCP> columnVec, const uint32_t sessionID = 0);
```
Comment (mariadb-LeonidFedorov) [inline dbcon/execplan/rowcolumn.h]: please remove

Comment (mariadb-LeonidFedorov) [reply dbcon/execplan/rowcolumn.h]: or implement

File: dbcon/execplan/constantcolumn.h (hunk 1 above)
Comment (mariadb-AndreyPiskunov) [reply dbcon/execplan/constantcolumn.h]: Removed

File: dbcon/execplan/filter.h (hunk 2 above)
Comment (mariadb-AndreyPiskunov) [reply dbcon/execplan/filter.h]: No, I removed all unnecessary includes.

File: dbcon/execplan/rowcolumn.h (hunk 3 above)
Comment (mariadb-AndreyPiskunov) [reply dbcon/execplan/rowcolumn.h]: I chose to implement.

File: dbcon/execplan/outerjoinonfilter.cpp [hunk 4]
```diff
@@ -66,6 +66,17 @@ const string OuterJoinOnFilter::toString() const
...
 
+string OuterJoinOnFilter::toCppCode(IncludeSet& includes) const
+{
+  includes.insert("outerjoinonfilter.h");
+  stringstream ss;
+
+  ss << "OuterJoinOnFilter(boost::shared_ptr<ParseTree>(new " << fPt->toCppCode(includes) << "))";
+
+  return ss.str();
+
```
Comment (mariadb-LeonidFedorov) [inline dbcon/execplan/outerjoinonfilter.cpp]: NIT: extra empty string

**Comment by mariadb-LeonidFedorov** [comment]
Please check and fix clang build errors: It's Debian and Ubuntu 20 in builds.
Also we see test001 is not passing here( why? ) logs show difference in SUBSTRING_INDEX.sql regression test

**Review by mariadb-SergeyZefirov — APPROVED** [review]
I like the result very much.

File: dbcon/execplan/outerjoinonfilter.cpp (hunk 4 above)
Comment (mariadb-AndreyPiskunov) [reply dbcon/execplan/outerjoinonfilter.cpp]: removed

**Pull request merged**